src/redisearch_rs/inverted_index/tests/freqs_only.rs

```diff
@@ -13,6 +13,33 @@ use inverted_index::{
     Decoder, DecoderResult, Delta, Encoder, RSIndexResult, freqs_only::FreqsOnly,
 };
 
+#[unsafe(no_mangle)]
+pub extern "C" fn ResultMetrics_Free(result: *mut RSIndexResult) {
```

Comment:
I think it would make sense to have those factored out in a test_utils file.

I created one [here](https://github.com/RediSearch/RediSearch/commit/05aa7ba520e3bed4aac90beae1b7ad6fe6d27200#diff-aef120155ea469468fc8877df49d78ad9bf6f55c4c70a39e2c0619db2019d183) but it's still in a WIP branch unfortunately.

Reply:
Would the idea be to make it easier to not repeat in all the tests? I can look into that. It just needs to be done carefully as some test files will want to override these functions for their own test cases.

Reply:
That makes sense. I'll see if there is a way that will allow the tests to provide their own implementation when needed. It should be possible.

Reply:
Done in https://github.com/RediSearch/RediSearch/pull/6496/commits/2254b0e9391141a39113a980757139bfe4e4b837

The down side is that pulling in the `c_mocks` module pulls in all the definitions and it is not possible to select only the ones needed. This means any tests which needs a custom definition will need to re-define all of them (and not pull in the `c_mocks` module)

Reply:
I still think it's valuable as I expect most tests will share the same definitions, right?